src/openzeppelin/token/erc721/dual721.cairo

```diff
@@ -0,0 +1,218 @@
+use openzeppelin::token::erc721::interface::IERC721;
+use openzeppelin::utils::try_selector_with_fallback;
+use openzeppelin::utils::Felt252TryIntoBool;
+use openzeppelin::utils::BoolIntoFelt252;
+use openzeppelin::utils::constants;
+
+#[derive(Copy, Drop)]
+struct DualCaseERC721 {
+    contract_address: ContractAddress
+}
+
+trait DualCaseERC721Trait {
+    fn name(self: @DualCaseERC721) -> felt252;
+    fn symbol(self: @DualCaseERC721) -> felt252;
+    fn token_uri(self: @DualCaseERC721, token_id: u256) -> felt252;
+    fn balance_of(self: @DualCaseERC721, account: ContractAddress) -> u256;
+    fn owner_of(self: @DualCaseERC721, token_id: u256) -> ContractAddress;
+    fn get_approved(self: @DualCaseERC721, token_id: u256) -> ContractAddress;
+    fn approve(self: @DualCaseERC721, to: ContractAddress, token_id: u256);
+    fn set_approval_for_all(self: @DualCaseERC721, operator: ContractAddress, approved: bool);
+    fn transfer_from(
+        self: @DualCaseERC721, from: ContractAddress, to: ContractAddress, token_id: u256
+    );
+    fn is_approved_for_all(
+        self: @DualCaseERC721, owner: ContractAddress, operator: ContractAddress
+    ) -> bool;
+    fn safe_transfer_from(
+        self: @DualCaseERC721,
+        from: ContractAddress,
+        to: ContractAddress,
+        token_id: u256,
+        data: Span<felt252>
+    );
+}
+
+impl DualCaseERC721Impl of DualCaseERC721Trait {
+    fn name(self: @DualCaseERC721) -> felt252 {
+        *call_contract_syscall(
+            *self.contract_address, constants::NAME_SELECTOR, ArrayTrait::new().span()
+        )
+            .unwrap_syscall()
+            .at(0)
+    }
+
+    fn symbol(self: @DualCaseERC721) -> felt252 {
+        *call_contract_syscall(
+            *self.contract_address, constants::SYMBOL_SELECTOR, ArrayTrait::new().span()
+        )
+            .unwrap_syscall()
+            .at(0)
+    }
+
+    fn token_uri(self: @DualCaseERC721, token_id: u256) -> felt252 {
+        let snake_selector = constants::TOKEN_URI_SELECTOR;
+        let camel_selector = constants::TOKENURI_SELECTOR;
+
+        let mut args = ArrayTrait::new();
+        args.append(token_id.low.into());
+        args.append(token_id.high.into());
+
+        *try_selector_with_fallback(
+            *self.contract_address, snake_selector, camel_selector, args.span()
+        )
+            .unwrap_syscall()
+            .at(0)
+    }
+
+    fn balance_of(self: @DualCaseERC721, account: ContractAddress) -> u256 {
+        let snake_selector = constants::BALANCE_OF_SELECTOR;
+        let camel_selector = constants::BALANCEOF_SELECTOR;
+
+        let mut args = ArrayTrait::new();
+        args.append(account.into());
+
+        let res = try_selector_with_fallback(
+            *self.contract_address, snake_selector, camel_selector, args.span()
+        )
+            .unwrap_syscall();
+
+        u256 { low: (*res.at(0)).try_into().unwrap(), high: (*res.at(1)).try_into().unwrap(),  }
```

Comment:
I might create a `unwrap_u256` util in the erc20 PR to abstract some of this away. A reminder for me :)